### PR TITLE
Corrections in dem parameter handler

### PR DESCRIPTION
**Pull request by shahabgol**
- Change the parameter name "density" to "density particle" in DEM to avoid possible interference with CFD parameters
- Define multiple particle types, floating walls and moving boundaries using loops in parameter handler
- Move unnecessary headers from dem.h to dem.cc

### CODE REVIEW

**Comment by blaisb** [comment]
Can you rebase @shahabgol  ? :)

**Comment by shahabgol** [comment]
> Can you rebase @shahabgol ? :)

Done! :)

**Comment by blaisb** [comment]
@shahabgol Can you update the insert_llist test to pass with your syntax change? :)

File: applications_tests/dem_2d/two_types_packing_in_circle.prm
Comment (blaisb) [inline applications_tests/dem_2d/two_types_packing_in_circle.prm:51]: Question that comes to mind is should it be "particle" or "particles" plural. Since it refers to the particles. It's not a big deal, but anyway.

File: include/core/parameters_lagrangian.h
Comment (blaisb) [inline include/core/parameters_lagrangian.h:263]:
```suggestion
      unsigned int max_number_floating_walls = 9;
```

**Review by blaisb — APPROVED** [review]
minor comments but this is good to go. I'll merge it after merging #181

File: applications_tests/dem_2d/two_types_packing_in_circle.prm
Comment (shahabgol) [reply applications_tests/dem_2d/two_types_packing_in_circle.prm]: Corrected

**Comment by shahabgol** [comment]
> @shahabgol Can you update the insert_llist test to pass with your syntax change? :)

The comments are applied.

**Review by blaisb — COMMENTED** [review]
Can you update the files in lethe/examples?
Once you do that, i'll merge

**Comment by shahabgol** [comment]
> @shahabgol it's very good. Can you update the files in lethe/examples?
> Once you do that, i'll merge

Done! :)

**Review by blaisb — APPROVED** [review]

**Pull request merged**
